decoder/label_schema_test.go

```diff
@@ -0,0 +1,175 @@
+package decoder
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func TestCandidatesAtPos_prefillRequiredBlocks(t *testing.T) {
+	tests := []struct {
+		name                      string
+		fileName                  string
```

Comment:
nitpick: I think the filename doesn't matter in the context of testing this feature, so you could probably just hard-code it below instead of duplicating it in every test case.